### DISCUSSION TITLE
[Experience sharing] Integration with PC1616 over MQTT

### DISCUSSION THREAD

**Discussion by witek1308** [taligentx/dscKeybusInterface#48] [dscKeybusInterface Discussion]
Hi,
I'm going to use this great stuff with PC1616 over MQTT (mosquitto broker) to OpenHAB2, waiting for NodeMCU to arrive.
MQTT should be straightforward but maybe there is something I should know about: specific settings or issues? Is there anything to know not to reinvent the wheel?

**Comment by witek1308** [comment]
What I observed is that when client was set to QoS(1) it was causing broker to retain messages (in my case disarm) which was overriding any arm. When I have some time to debug, will try to update but if I'm not too quick in my judgement this would suggest the way it is handled in the code and not sending ack for this type of message.

**Comment by taligentx** [comment]
Hi @witek1308 - is OpenHAB sending commands to the 'dsc/Set' topic with retain?  It should be one-shot to make sure the interface doesn't repeat the command to the panel.

The retain flag is handy for the status topics in case the microcontroller is restarted.

**Comment by witek1308** [comment]
Hi it was long ago so don't remember details but definitely I had to 
configure separate mqtt profile on for dsc to get rid if the retain. I 
don't remember though if it was because of the Arduino code or maybe OH2 
configuration which for me defaults to retain as most of my devices send 
messages which need to be retained

--- Wiadomość oryginalna ---
Od: Nikhil Choudhary notifications@github.com
Data: 22 maja 2019 00:01:42
Do: taligentx/dscKeybusInterface dscKeybusInterface@noreply.github.com
DW: witek1308 witek_1308@poczta.onet.pl, Mention mention@noreply.github.com
Temat: Re: [taligentx/dscKeybusInterface] [Experience sharing] Integration 
with PC1616 over MQTT (#48)

**Comment by taligentx** [comment]
Gotcha, thanks for the info - I'll keep an eye out to see if there are further issues with having the status messages set to retain.
